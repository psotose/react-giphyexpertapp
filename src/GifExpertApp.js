import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

  // const categories = ['One Punch', 'Samurai X', 'Dragon Ball']
  const [categories, setCategories] = useState(['One Punch']);

  // const handleAdd = () => {
  //   // setCategories( [...categories, 'HunterXHunter'] );
  //   setCategories( categories => [...categories, 'HunterXHunter' ] );
  // };

  return (
    <div>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories= { setCategories }/>  
      <hr />

      {/* <button onClick={ handleAdd } >Agregar</button> */}
      <ol>
        {
          categories.map( category =>
            <GifGrid 
              key={category}
              category={ category } 
            />
            // {return <li key={ category }> { category } </li>} si le saco las llaves de afuera y el return, lo retorna =
          )
        }
      </ol>
    </div>
  )
}



