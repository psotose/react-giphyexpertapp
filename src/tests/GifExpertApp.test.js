import React from 'react'
import { shallow } from 'enzyme';
import { GifExpertApp } from "../GifExpertApp";

describe('Testing <GifExpertApp />', () => {

  test('should be render correctly', () => {
    const wrapper =  shallow(<GifExpertApp />);
    expect ( wrapper ).toMatchSnapshot();
  });

  test('should show a category list', () => {

    const categories = ['One Punch', 'DragonBall'];
    const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

    expect ( wrapper ).toMatchSnapshot();
    expect ( wrapper.find('GifGrid').length ).toBe( categories.length );
  });

});