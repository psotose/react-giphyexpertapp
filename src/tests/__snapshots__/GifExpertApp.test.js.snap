// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifExpertApp /> should be render correctly 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`Testing <GifExpertApp /> should show a category list 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="DragonBall"
      key="DragonBall"
    />
  </ol>
</div>
`;
